applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx

```diff
@@ -1,33 +1,175 @@
 /* eslint-disable react/jsx-key */
+import { useEffect, useRef, useState } from 'react'
+
 import Text from '../../Text'
 import t from '../../../locales'
 import Button from '../../Button'
+
 import { useAppDispatch } from '../../../store/hooks'
 import { setOnboardingComplete } from '../../../store/app'
+import { actions as baseNodeActions } from '../../../store/baseNode'
+import {
+  CtaButtonContainer,
+  FlexContent,
+  ProgressContainer,
+  RemainingTime,
+} from './styles'
+import ProgressBar from '../../ProgressBar'
+import { TBotMessage, TBotMessageHOCProps } from '../../TBot/TBotPrompt/types'
+import { useTheme } from 'styled-components'
 
 /**
  * @TODO - how the Blockchain synchronization should to work?
  */
-const messages = [
-  () => {
-    const dispatch = useAppDispatch()
-
-    return (
-      <>
-        <Text as='span' type='defaultHeavy'>
-          {t.onboarding.lastSteps.message1} ✨💪
+
+const Progress = ({ progress, time }: { progress: number; time: string }) => {
+  const theme = useTheme()
+
+  return (
+    <ProgressContainer>
+      <Text
+        color={theme.accent}
+        type='smallMedium'
+        style={{
+          alignSelf: 'flex-start',
+          marginBottom: theme.spacingVertical(0.5),
+        }}
+      >
+        {t.onboarding.lastSteps.blockchainIsSyncing}
+      </Text>
+      <ProgressBar value={progress} />
+      <RemainingTime>
+        <Text type='smallMedium'>
+          {time} {t.common.adjectives.remaining.toLowerCase()}
         </Text>
-        <Button
-          variant='button-in-text'
-          onClick={() => dispatch(setOnboardingComplete(true))}
-        >
-          <Text as='span' type='defaultUnder'>
-            Exit data synchronization
-          </Text>
-        </Button>
-      </>
-    )
-  },
-]
-
-export default messages
+      </RemainingTime>
+    </ProgressContainer>
+  )
+}
+
+export const BlockchainSyncStep = ({
+  pushMessages,
+  updateMessageBoxSize,
+}: {
+  pushMessages: (msgs: TBotMessage[]) => void
+} & TBotMessageHOCProps) => {
+  const dispatch = useAppDispatch()
+
+  const contentRef = useRef<HTMLDivElement | null>(null)
+
+  const [error, setError] = useState(false)
+  const [syncStarted, setSyncStarted] = useState(false)
+  const [syncFinished, setSyncFinished] = useState(false)
+  const [progress, setProgress] = useState(0)
+  const [remainingTime, setRemainingTime] = useState('55 min')
+
+  const startSync = async () => {
+    setSyncStarted(true)
+    pushMessages([
+      {
+        content: <Text>{t.onboarding.lastSteps.message2}</Text>,
+        barFill: 0.875,
+        noSkip: true,
+      },
+    ])
+    try {
+      await dispatch(baseNodeActions.startNode()).unwrap()
+    } catch (e) {
+      setError(true)
+    }
+  }
+
+  useEffect(() => {
+    if (error) {
```

Comment:
it's not necessary to depend on state and react to it in useEffect

make this a function and call it instead of `setError(true)`